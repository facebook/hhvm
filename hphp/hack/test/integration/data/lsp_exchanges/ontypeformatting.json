[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "method": "initialize",
    "params": {
      "initializationOptions": {},
      "processId": null,
      "rootPath": "${root_path}",
      "capabilities": {}
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/didOpen",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}",
        "languageId": "hack",
        "version": 1,
        "text": "${php_file}"
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 11,
    "method": "textDocument/onTypeFormatting",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 9,
        "character": 58
      },
      "ch": ";",
      "options": {
        "tabSize": 2,
        "insertSpaces": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "method": "textDocument/onTypeFormatting",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 13,
        "character": 1
      },
      "ch": "}",
      "comment": "NOTE: This tests bracket-matching",
      "comment": "The position describes that of the cursor position after typing '{' (AKA the cursor space in between '{' and '}'), not that of the closing brace '}'",
      "options": {
        "tabSize": 2,
        "insertSpaces": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 13,
    "method": "textDocument/onTypeFormatting",
    "params": {
      "textDocument": {
        "uri": "${php_file_uri}"
      },
      "position": {
        "line": 15,
        "character": 16
      },
      "ch": "}",
      "options": {
        "tabSize": 2,
        "insertSpaces": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "method": "shutdown",
    "params": {}
  }
]
