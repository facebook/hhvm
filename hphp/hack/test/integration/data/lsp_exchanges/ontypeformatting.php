<?hh

function test_otf(mixed ...$_): void {
}

function otf_1() {
  test_otf(
    '1234567890',
    '1234567890',
    '1234567890','1234567890', '1234567890','1234567890'); // 11: Format on ";"
}


{}     // 12: Format on bracket-matching "{}"

function otf(){}      // 13: Format on "}"
