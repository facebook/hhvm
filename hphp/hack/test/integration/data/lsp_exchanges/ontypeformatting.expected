[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<",
            "[",
            "'",
            "\""
          ]
        },
        "signatureHelpProvider": {
          "triggerCharacters": [
            "(",
            ","
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "documentOnTypeFormattingProvider": {
          "firstTriggerCharacter": ";",
          "moreTriggerCharacter": [
            "}"
          ]
        },
        "renameProvider": true,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 11,
    "result": [
      {
        "range": {
          "start": {
            "line": 5,
            "character": 17
          },
          "end": {
            "line": 9,
            "character": 58
          }
        },
        "newText": "{\n  test_otf(\n    '1234567890',\n    '1234567890',\n    '1234567890',\n    '1234567890',\n    '1234567890',\n    '1234567890',\n  );"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 12,
    "result": [
      {
        "range": {
          "start": {
            "line": 13,
            "character": 0
          },
          "end": {
            "line": 13,
            "character": 1
          }
        },
        "newText": "{"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 13,
    "result": [
      {
        "range": {
          "start": {
            "line": 15,
            "character": 0
          },
          "end": {
            "line": 15,
            "character": 16
          }
        },
        "newText": "function otf() {}"
      }
    ]
  },
  {
    "jsonrpc": "2.0",
    "id": 999,
    "result": null
  }
]